src/crypto/rsa.rs

```diff
@@ -46,13 +61,11 @@ impl<T: Rng> Padding for Pkcs1Padding<T> {
         let ps_len = k - 3 - input_len;
 
         for i in 0..ps_len {
-            let x = loop {
-                match self.rng.gen::<u8>() {
-                    0x00 => continue,
-                    x => break x,
-                }
-            };
-            output[i + 2] = x;
+            while output[i + 2] == 0 {
```

Comment:
An optimization would be to first run `self.rng.fill(&mut output[2..pslen+2])` before this loop